### PR TITLE
Opt-in rules for loader include/exclude

### PR DESCRIPTION
**Pull request by jonathanewerner**
Implements #38. Two opt-in rules for having stricter "best-practice" validations for loader include/exclude options. See README for explanation.

cc @bebraw as you brought this issue up. :)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -13,6 +13,18 @@
 ### Testing workflow
 Have a look at `test/README.md` to see how testing works. Run `npm run watch:test` while developing. Have fun! :)
 
+### Tips for working with Joi
+- Use `!!` at the beginning of your custom error string to get rid of the key at the beginning of the error message. Example:
```
Comment (jonathanewerner) [inline CONTRIBUTING.md]: This is actually awesome sauce; why did I not know about this before? :)
(Means that we can get rid of pesky error messages that look  like "`"value" bla bla`", changing them to "`bla bla`").

File: README.md [hunk 2]
```diff
@@ -79,6 +79,8 @@ module.exports = validate(config, { schemaExtension: yourSchemaExtension })
 #### Rules
 Some validations do more than just validating your data shape, they check for best practices and do "more" which you might want to opt out of / in to. This is an overview of the available rules (we just started with this, this list will grow :)):
 - **no-root-files-node-modules-nameclash** (default: true): this checks that files/folders that are found in directories specified via webpacks `resolve.root` option do not nameclash with `node_modules` packages. This prevents nasty path resolving bugs (for a motivating example, have a look at [this redux issue](https://github.com/reactjs/redux/issues/1681)).
+- **loader-enforce-include-or-exclude** (default: false): enforce that [loader](https://webpack.github.io/docs/configuration.html#module-loaders) objects use `include` or/and `exclude`, throw when neither is supplied. Without supplying one of these conditions it is too easy to process too many files, for example your `node_modules` folder.
```
Comment (jonathanewerner) [inline README.md]: I'd like your input if you dig these rule names + their descriptions.

File: README.md [hunk 3]
```diff
@@ -79,6 +79,8 @@ module.exports = validate(config, { schemaExtension: yourSchemaExtension })
...
+- **loader-prefer-include** (default: false): enforce that [loader](https://webpack.github.io/docs/configuration.html#module-loaders) objects use `include` and not `exclude`. `exclude` makes it easy to match too many files, which might inadvertently slow your build down.
```
Comment (jonathanewerner) [inline README.md]: I'd like your input what you think of these rules, their names + their descriptions/rationales.

File: src/properties/module/index.js [hunk 4]
```diff
@@ -1,36 +1,86 @@
 import Joi from 'joi'
+import { oneLine } from 'common-tags'
```
Comment (jonathanewerner) [inline src/properties/module/index.js]: `common-tags` seems sweet, wanted to try it out here.

File: src/properties/module/index.js [hunk 5]
```diff
@@ -1,36 +1,86 @@
...
+    include: conditionSchema,
+    loader: Joi.string(),
+    query: Joi.object(),
+    loaders: Joi.array().items(Joi.string()),
+  })
+    .xor('loaders', 'loader')
+    .nand('loaders', 'query')
+    .options({ language: { object: { nand: LOADERS_QUERY_MESSAGE } } })
+
+  if (rules['loader-enforce-include-or-exclude']) {
```
Comment (jonathanewerner) [inline src/properties/module/index.js]: So I guess this is how the general pattern of applying rules to the schema will look like. If you come up with a better pattern, please don't hesitate. :)

File: README.md (hunk 3 above)
Comment (bebraw) [inline README.md]: Naming looks good to me. If we can mimic solutoins like ESLint here, that will work.

A lot of existing configurations could be "broken", yeah.

**Comment by bebraw** [comment]
LGTM. 👍

**Comment by jonathanewerner** [comment]
Sweet. Let's merge this then.

**Pull request merged**